patches/BUNDLE-2554__set_payment_info_bug_fix__3.4.1.patch

```diff
@@ -0,0 +1,126 @@
+             ];
+
+             $responseParser = $this->clientFactory->create($storeId)->setOrderReferenceDetails($data);
+-            try {
+-                $response       = $this->amazonSetOrderDetailsResponseFactory->create(
+-                    [
+-                    'response' => $responseParser
+-                    ]
+-                );
++            $response = $this->amazonSetOrderDetailsResponseFactory->create([
++                'response' => $responseParser
++            ]);
+
+-                $this->validateConstraints($response, $allowedConstraints);
+-            } catch (AmazonServiceUnavailableException $e) {
+-                if($e->getApiErrorCode() == 'OrderReferenceNotModifiable') {
+-                    $this->logger->warning(
+-                        "Could not modify Amazon order details for $amazonOrderReferenceId: "
+-                        . $e->getApiErrorMessage()
+-                    );
+-                } else {
+-                    throw $e;
+-                }
+-            }
++            $this->validateConstraints($response, $allowedConstraints);
+         } catch (LocalizedException $e) {
+             throw $e;
+         } catch (Exception $e) {
+diff --git a/vendor/amzn/amazon-pay-module/Plugin/ConfirmOrderReference.php b/vendor/amzn/amazon-pay-module/Plugin/ConfirmOrderReference.php
+index 6bad468..14b1264 100644
+--- a/vendor/amzn/amazon-pay-module/Plugin/ConfirmOrderReference.php
++++ b/vendor/amzn/amazon-pay-module/Plugin/ConfirmOrderReference.php
+@@ -25,6 +25,7 @@
+ use Amazon\Payment\Model\OrderInformationManagement;
+ use Magento\Quote\Api\Data\PaymentInterface;
+ use Magento\Quote\Api\Data\AddressInterface;
++use Magento\Framework\Webapi\Rest\Request;
+ use Magento\Framework\Exception\LocalizedException;
+ use Amazon\Payment\Gateway\Config\Config as GatewayConfig;
+ use Magento\Quote\Api\CartRepositoryInterface;
+@@ -41,6 +42,11 @@ class ConfirmOrderReference
+      */
+     private $checkoutSession;
+
++    /**
++     * @var Request
++     */
++    private $request;
++
+     /**
+      * @var OrderInformationManagement
+      */
+@@ -54,19 +60,31 @@ class ConfirmOrderReference
+     /**
+      * ConfirmOrderReference constructor.
+      * @param Session $checkoutSession
++     * @param Request $request
+      * @param OrderInformationManagement $orderInformationManagement
+      * @param CartRepositoryInterface $quoteRepository
+      */
+     public function __construct(
+         Session $checkoutSession,
++        Request $request,
+         OrderInformationManagement $orderInformationManagement,
+         CartRepositoryInterface $quoteRepository
+     ) {
+         $this->checkoutSession = $checkoutSession;
++        $this->request = $request;
+         $this->orderInformationManagement = $orderInformationManagement;
+         $this->quoteRepository = $quoteRepository;
+     }
+
++    /**
++     * @return boolean
++     */
++    protected function canConfirmOrderReference()
```

Comment:
use private

Reply:
This is a fix for 3rd-party extension, can be ignored in favor of the original change